crates/node/src/types.rs

```diff
@@ -32,22 +32,41 @@ pub struct TransactionRejection {
 #[derive(Debug, Deserialize, Serialize)]
 #[non_exhaustive]
 pub enum CatchupStreamMessage {
-    Request(CatchupRequest),
-    TransactionsBatch(CatchupTransactionBatch),
+    ApplicationBlobRequest(CatchupApplicationBlobRequest),
+    ApplicationBlobSize(CatchupApplicationBlobSize),
+    ApplicationBlobChunk(CatchupApplicationBlobChunk),
+    TransactionsRequest(CatchupTransactionsRequest),
+    TransactionsBatch(CatchupTransactionsBatch),
     Error(CatchupError),
 }
 
 #[derive(Debug, Deserialize, Serialize)]
 #[non_exhaustive]
-pub struct CatchupRequest {
+pub struct CatchupApplicationBlobRequest {
+    pub context_id: ApplicationId,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct CatchupApplicationBlobSize {
+    pub size: u64,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[non_exhaustive]
+pub struct CatchupApplicationBlobChunk(Box<[u8]>);
```

Comment:
unless you want to employ some additional chunking, it's worth noting that blobstore chunks are 1MiB in size

https://github.com/calimero-network/core/blob/64420b00cf96c27d3674a5810357f70a8a94332f/crates/store/blobs/src/lib.rs#L24

so this will be encoded as an array of 1,048,576 elements, which isn't the best way to send it over the wire

which is fine for now, but I can also see that `Message::new` just takes in a `Vec<u8>`, so maybe we can change it to borsh serialization instead to be more compact

Eventually, it'll be good to use a `Cow<'a, [u8]>` instead of a `Box<u8>` to reduce the expectations from the constructor, since borsh serialization only needs a reference, but also fine for now, just noting the parts we'd have to employ some care in